src/main/java/entities/DesignableMaze.java

```diff
@@ -13,11 +20,69 @@ public void deleteWall(int row, int col) {
         this.state[row][col] = Maze.ENCODING.get("empty");
     }
 
+    public void placeStart(int row, int col) {
+        this.state[startLocation[0]][startLocation[1]] = Maze.ENCODING.get("empty");
+
+        this.state[row][col] = Maze.ENCODING.get("start");
+        startLocation[0] = row;
+        startLocation[1] = col;
+    }
+    public void placeEnd(int row, int col) {
+        this.state[endLocation[0]][endLocation[1]] = Maze.ENCODING.get("empty");
+
+        this.state[row][col] = Maze.ENCODING.get("end");
+        endLocation[0] = row;
+        endLocation[1] = col;
+    }
+
+    public char getCell(int row, int col){
+        return this.state[row][col];
+    }
+
     public void fill() {
         for (int i = 0; i < this.getNumRow(); i++) {
             for (int j = 0; j < this.getNumCol(); j++) {
                 placeWall(i, j);
             }
         }
     }
+
+    public void setButtonArray(Button[][] buttonarray){
+        for (int i = 0; i < buttonarray.length; i++) {
+            for (int j = 0; j < buttonarray[0].length; j++) {
+                if(buttonarray[i][j] == null){
+                    buttonarray[i][j] = new Button();
+                }
+                if(getCell(i, j) == '#'){
+                    buttonarray[i][j].setText("#");
+                } else if (getCell(i, j) == '.'){
+                    buttonarray[i][j].setText(".");
+                } else if (getCell(i, j) == 'S'){
+                    buttonarray[i][j].setText("S");
+                } else {
+                    buttonarray[i][j].setText("E");
+                }
+            }
+        }
+    }
+
```

Comment:
I don't think this entity should be allowed to use JavaFx, since that would mean the entity relies on the outermost layer (UI frameworks).